### DISCUSSION TITLE
How to use custom variants with v3

### DISCUSSION THREAD

**Discussion by Daidalos117** [chakra-ui/chakra-ui#9026] [Q&A]
Hello all 👋,
I am migrating from v2 to v3. Before I had something like this:
```
export const buttonStyles = defineStyleConfig({
  variants: {
    primary: {
     color:  "red"
   }
}})
```
And I could use it like this:
```
import { Button } from "@chakra-ui/react";
<Button variant="primary">Hi</Button>
```

Changed it to

```
defineRecipe({
  variants: {
    primary: {
      true: {
        color: "red",
      }
    }
  }
}
```
But when I import the Button component from chakra-ui, the variant doesn't work
```
import { Button } from "@chakra-ui/react";
<Button _primary_>Hi</Button>
```

Regarding the documentation, I can do something like this:
```
export const Button = chakra("button", buttonRecipe)
<Button primary>Hi</Button>
```
Is it really the pattern I should be using?

Thanks in advance! 🙏

**Comment by segunadebayo** [comment]
You probably need to update the typings for the theme using the Chakra CLI
https://www.chakra-ui.com/docs/get-started/cli#chakra-typegen

**Reply by Daidalos117** [reply]
Already tried that numerous times. I replicated it on [StackBlitz](https://stackblitz.com/edit/chakra-ui-v3-4ymrwm?file=src%2FApp.tsx). Even tho I cannot regenerate typing there, the variant itself should be working if that's correct way to use it, right?

**Comment by Daidalos117** [comment]
Found the solution:
https://github.com/chakra-ui/chakra-ui/discussions/9016#discussioncomment-11174648
